04 Strategy Library/118 Time Series Momentum Effect/02 Method.html

```diff
@@ -0,0 +1,82 @@
+<p>
+  As the strategy needs the continuous futures contract, we import the custom data from Quandl.
+  We create a universe of tradable commodity futures from liquid commodity futures traded on CME and ICE and CBOT, you could add more using Quandl API.
+  They are all liquid and active financial derivatives. The data from Quandl are non-adjusted price based on spot-month continuous contract calculations.
+  The data resolution is daily.
+</p>
+<p>
+  The first step is importing the data.
+</p>
+<div class="section-example-container">
+<pre class="python">from QuantConnect.Python import PythonQuandl
+for symbol in self.symbols_Settle:
+    self.AddData(QuandlFutures, symbol, Resolution.Daily)
+
+class QuandlFutures(PythonQuandl):
+    def __init__(self):
+        self.ValueColumnName = "Settle"
+</pre>
+</div>
+<p>
+  Here we use the indicator <code>RateOfChange(period)</code> to simulate the momentum return. Here the period is 12 months.
+  As we are using the custom data, the indicator initialization should use the history request to update the value manually.
+  All indicators are saved in the dictionary <code>self.roc</code>.
+</p>
+<div class="section-example-container">
+<pre class="python">self.roc = {}
+for symbol in self.symbols:
+    self.AddData(QuandlFutures, symbol, Resolution.Daily)
+    self.roc[symbol] = RateOfChange(period)
+    hist = self.History([symbol], 300, Resolution.Daily).loc[symbol]
+    for i in hist.itertuples():
+        self.roc[symbol].Update(i.Index, i.value)
+</pre>
+</div>
+<p>
+  In <code>OnData(self, data)</code>, indicators for all futures contracts are updated every day with the settlement price.
+</p>
+<p>
+  We choose instruments with positive 12-month momentum to long, and instrucments with negative 12-month momentum to short.
```

Comment:
"We take a long position in instruments with positive 12-month momentum and short positions with negative momentum."